src/main/java/net/sf/rails/game/special/SpecialSingleTileLay.java

```diff
@@ -0,0 +1,164 @@
+            throw new ConfigurationException("<SpecialSingleTileLay> tag missing");
+        }
+
+        locationCodes = tileLayTag.getAttributeAsString("location");
+        if (!Util.hasValue(locationCodes))
+            throw new ConfigurationException("SpecialSingleTileLay: location missing");
+
+        tileId = tileLayTag.getAttributeAsString("tile", null);
+
+        String coloursString = tileLayTag.getAttributeAsString("colour");
+        if (Util.hasValue(coloursString)) {
+            tileColours = coloursString.split(",");
+        }
+
+        name = tileLayTag.getAttributeAsString("name");
+
+        extra = tileLayTag.getAttributeAsBoolean("extra", extra);
+        free = tileLayTag.getAttributeAsBoolean("free", free);
+        connected = tileLayTag.getAttributeAsBoolean("connected", connected);
+        discount = tileLayTag.getAttributeAsInteger("discount", discount);
+
+        if (tileId != null) {
+            description = LocalText.getText("LayNamedTileInfo",
+                    tileId,
+                    name != null ? name : "",
+                            locationCodes,
+                            (extra ? LocalText.getText("extra"):LocalText.getText("notExtra")),
+                            (free ? LocalText.getText("noCost") : discount != 0 ? LocalText.getText("discount", discount) :
+                                LocalText.getText("normalCost")),
+                            (connected ? LocalText.getText("connected") : LocalText.getText("unconnected"))
+            );
+        } else {
+            description = LocalText.getText("LayTileInfo",
+                    locationCodes,
+                    (tileColours != null ? Arrays.toString(tileColours).replaceAll("[\\[\\]]", ""): ""),
+                    (extra ? LocalText.getText("extra"):LocalText.getText("notExtra")),
+                    (free ? LocalText.getText("noCost") : discount != 0 ? LocalText.getText("discount", discount) :
+                        LocalText.getText("normalCost")),
+                    (connected ? LocalText.getText("connected") : LocalText.getText("unconnected"))
+            );
+        }
+
+    }
+
+    @Override
+	public void finishConfiguration (RailsRoot root)
+    throws ConfigurationException {
+
+        TileManager tmgr = root.getTileManager();
+        MapManager mmgr = root.getMapManager();
+        MapHex hex;
+
+        if (tileId != null) {
+            tile = tmgr.getTile(tileId);
+        }
+
+        locations = new ArrayList<>();
+        for (String hexName : locationCodes.split(",")) {
+            hex = mmgr.getHex(hexName);
+            if (hex == null)
+                throw new ConfigurationException("Location " + hexName
+                        + " does not exist");
+            locations.add(hex);
+        }
+
+    }
+
+    @Override
+    public boolean isExecutionable() {
+        return true;
+    }
+
+    @Override
+    public boolean isExtra() {
+        return extra;
+    }
```

Comment:
If the fields are specified in the parent class it is better to also implement these getter methods in the parent class

Reply:
I do agree

Reply:
This would be my only comment too